javatests/arcs/core/storage/CapabilitiesResolverTest.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.storage
+
+import arcs.core.common.ArcId
+import arcs.core.data.Capabilities
+import arcs.core.data.Schema
+import arcs.core.storage.driver.*
+import arcs.core.testutil.assertThrows
+import arcs.jvm.storage.database.testutil.MockDatabaseFactory
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import java.lang.Error
+
+/** Tests for [CapabilitiesResolver]. */
+@RunWith(JUnit4::class)
+class CapabilitiesResolverTest{
+    @Before
+    fun setUp() {
+        VolatileDriverProvider(ArcId.newForTest("test"))
+    }
+
+    @After
+    fun tearDown() {
+        CapabilitiesResolver.reset()
+        CapabilitiesResolver.defaultCreators.clear()
+    }
+
+    @Test
+    fun capabilitiesResolver_createsStorageKeysDefault() {
+        val options = CapabilitiesResolver.StorageKeyOptions(ArcId.newForTest("test"))
+        val resolver = CapabilitiesResolver(options)
+        assertThat(resolver.findStorageKeyProtocols(Capabilities.TiedToArc)).containsExactly("volatile")
+        assertThat(resolver.findStorageKeyProtocols(Capabilities.TiedToRuntime).isEmpty()).isTrue()
+        assertThat(resolver.findStorageKeyProtocols(Capabilities.Persistent).isEmpty()).isTrue()
+        assertThat(resolver.createStorageKey(Capabilities.TiedToArc) is VolatileStorageKey).isTrue()
+        assertThrows(IllegalStateException::class) {
+            resolver.createStorageKey(Capabilities.TiedToRuntime)
+        }
+        assertThrows(IllegalStateException::class) {
+            resolver.createStorageKey(Capabilities.Persistent)
+        }
+        assertThrows(IllegalStateException::class) {
+            resolver.createStorageKey(Capabilities(setOf(Capabilities.Capability.TiedToArc, Capabilities.Capability.Persistent)))
+        }
+    }
+
+    @Test
+    fun capabilitiesResolver_createsStorageKeysCtor() {
+        val options = CapabilitiesResolver.StorageKeyOptions(ArcId.newForTest("test"))
+        val resolver = CapabilitiesResolver(options,
+            mutableMapOf("ramdisk" to CapabilitiesResolver.CapabilitiesCreator(
+                Capabilities.TiedToRuntime, { (arcId), _ -> RamDiskStorageKey(arcId.toString()) }
+            ))
```

Comment:
Nits: move the value to the next line, and add a line break before the closing paren.  Also, same nits apply here regarding moving the lambda outside of the function call parens, and not using destructing for `arcId`.